plugin/rename_file.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+from .core.open import open_file_uri
+from .core.protocol import Notification, RenameFilesParams, Request, WorkspaceEdit
+from .core.registry import LspWindowCommand
+from pathlib import Path
+from urllib.parse import urljoin
+import os
+import sublime
+import sublime_plugin
+
+
+class RenameFileInputHandler(sublime_plugin.TextInputHandler):
+    def want_event(self) -> bool:
+        return False
+
+    def __init__(self, file_name: str) -> None:
+        self.file_name = file_name
+
+    def name(self) -> str:
+        return "new_name"
+
+    def placeholder(self) -> str:
+        return self.file_name
+
+    def initial_text(self) -> str:
+        return self.placeholder()
+
+    def validate(self, name: str) -> bool:
+        return len(name) > 0
+
+
+class LspRenameFileCommand(LspWindowCommand):
+    capability = 'workspace.fileOperations.willRename'
+
+    def is_enabled(self):
+        return True
+
+    def want_event(self) -> bool:
+        return False
+
+    def input(self, args: dict) -> sublime_plugin.TextInputHandler | None:
+        if "new_name" in args:
+            return None
+        old_path = self.get_old_path(args.get('paths'), self.window.active_view())
+        return RenameFileInputHandler(Path(old_path).name)
+
+    def run(
+        self,
+        new_name: str,  # new_name can be: FILE_NAME.xy OR ./FILE_NAME.xy OR ../../FILE_NAME.xy
+        paths: list[str] | None = None,  # exist when invoked from the sidebar with "LSP: Rename"
```

Comment:
`paths` is passed implicitly when invoked form the command is invoked from the sidebar.
For more info read  the [Side Bar.sublime-menu](https://www.sublimetext.com/docs/menus.html#available-menus) section